Library/Homebrew/requirements/xcode_requirement.rb

```diff
@@ -11,8 +11,6 @@ class XcodeRequirement < Requirement
 
   attr_reader :version
 
-  satisfy(build_env: false) { xcode_installed_version }
```

Comment:
How about leaving this as-is and overriding `xcode_installed_version` to be always `true` only on Linux? That would save changing things on macOS at all.

Reply:
I also thought about that strategy. But it feels weird for `xcode_installed_version` to be called (and to be defined) on Linux. This is why I proposed to move the Mac specific code to the Mac class, to make the base class platform-agnostic. But I think we never used that strategy elsewhere.

I think the advantage of moving the `satisfy` part to the platform-specific classes will force the next developers that touch that code to not blindly modify the code but think about how it could impact the linux part.

Reply:
> But it feels weird for `xcode_installed_version` to be called (and to be defined) on Linux.

It feels weird to me that `depends_on :xcode` is part of formulae on Linux. There's a host of macOS-specific stuff on Linux and I think the override approach is nicer particularly in cases like this where the class doesn't really make sense at all on Linux.

Reply:
Let's go with that version. Both strategies sound weird, so I have no strong feeling about it. I changed the code accordingly.